src/lib.rs

```diff
@@ -72,6 +72,35 @@ impl io::Read for IoBuffer {
     }
 }
 
+impl IoBuffer {
+    /// Read a full line (terminated by `\n`), if any.  Any partial line is
+    /// left unread. The terminator is not included in the result.
+    pub fn read_full_line(&mut self) -> Option<Vec<u8>> {
```

Comment:
Would a nicer API be to `read_terminator` or something, and then this just becomes a special case with `\n`?  Might be overkill.

Reply:
Good suggestion, I might as well. Done.